Update namelist_defaults_cam.xml for EW mpasa grids

gdicker1 opened the pull request:
Add default values for mpasa120, mpasa60, and mpasa30 grids. These change should reduce the amount of commands needed to setup FHS94, FKESSLER, F2000climo, and QPC6 compsets.

Commit:
Add 60 and 30km ncdata files for analytic ICs w/o topography

These files have been generated using MPAS-A init-atmosphere core for
CAM-MPAS grids (case 13) based on standard 60 and 30km MPAS meshes.

Also add default values for mpasa60 and mpasa30 hgrids. Needed default
values for mpas_dt and mpas_len_disp since they are affected by hgrids.

Commit:
namelist_defaults files for mpasa120, 60, and 30 grids

These modifications should let QPC6 and F2000climo test cases to run by
default for these grids.

Added values for:
- ncdata for 32 level cases starting 01-01-2000T00:00:00Z
- bnd_topo
- drydep_srf_file

Commit:
Add ic_ymd to 01-01-2000 mpas data files

Commit:
Ensure mpasa120 aquaplanet doesn't use topo ncdata

Adds redundant entry to list of ncdata since QPC6 cases using mpasa120
grids mistakenly use mpasa120_L32_topo_coords_c201022.nc

Commit:
Use analytic_ic for aquaplanet mpasa grids

This is a simplification specific to EarthWorks. All QPC6/aquaplanet
compsets on mpasa hgrids also use
analytic_ic_type='us_standard_atmosphere'

dazlich reviewed (commented):
I was giving this a test at 30km and the file /glade/p/cesmdata/inputdata/atm/cam/inic/mpas/cami_01-01-2000_00Z_mpasa30_L32_CFSR_230302.nc didn't exist. The bnd_topo and dry_dep files looks to be there.

gdicker1 commented:
@dazlich, understood. I copied that file from the /glade/p/univ/ucsu0085/inputdata and the [F2000 runscript](https://github.com/NCAR/Earthworks_scripts/blob/fcffbcdcfee48af609c8285559ecfcc2221bd15e/EWv1_CreateBuildRun/F2000climo_cheyenne_CBR.sh#L97), so I don't think we can expect that to be in the cesmdata dir or server.

If you point your case to "/glade/p/ucsu0085/inputdata2" (either with `./xmlchange DIN_LOC_ROOT=...` or using the `--input-dir "..."` flag with `create_newcase`) does it work?

dazlich commented:
Ah, got it @gdicker1.  I've modified my scripts and all is good. Somewhere along the way I missed that inputdata2 directory.

dazlich reviewed (approved):
Good to go!

gdicker1 commented:
That's because it's new (today)! It might not have everything (esp. for FullyCoupled), but I'm going to keep adding to it and adjusting our externals so hopefully it will in the coming weeks.

dazlich commented:
Great! I’ll add things for fully coupled there as I work with it. I’ll be adjusting ocean and seaice entries where they build their streams.

On Sep 1, 2023, at 12:42 PM, gdicker1 ***@***.***> wrote:


** Caution: EXTERNAL Sender **


That's because it's new (today)! It might not have everything (esp. for FullyCoupled), but I'm going to keep adding to it and adjusting our externals so hopefully it will in the coming weeks.

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/CAM/pull/4#issuecomment-1703183174>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWH7M3CH6LMNIY3Y5CLXYIUDDANCNFSM6AAAAAA4DTKA4Y>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

dazlich reviewed (commented):
@gdicker1 , could you add the following line to the namelist_defaults_cam.xml file?

<ncdata hgrid="mpasa120" nlev="58" ic_ymd="20000101" >atm/cam/inic/mpas/x1.40962.pel.58levels_2000010100_init.nc</ncdata>

I've already copied the file over to inputdata2.

Later, I'll make a similar request for a 58L 15km file I have.

I've been working with ocn and ice files, but I can't copy them into their directories, you're the only one with write permission.

dazlich reviewed (commented):
Can you also add this line to the namelist_defaults_cam.xml file?

<bnd_topo hgrid="mpasa120" nlev="58" >atm/cam/topo/mpasa120_gmted2010_modis_bedmachine_nc3000_Laplace0100_20220728.nc</bnd_topo>

The 58L initial condition file I added was generated with this topo file.

Commit:
Add 120km ncdata and bnd_topo file for 58 levels

Files generated by Don Dazlich for use in EarthWorks compsets

gdicker1 commented:
@dazlich I've added the ncdata and bnd_topo files you requested but I haven't pushed it yet. I'd like some context on x1.40962.pel.58levels_2000010100_init.nc and to maybe align it better with other ncdata files already in the namelist_defaults.xml.

- What's the significance of "pel" in the file name? "PE layout"?
- Did you use CFSR data to initialize or some other data source?

Given that, would "cami_01-01-2000_00Z_mpasa120_L58_CFSR_230901.nc" be an acceptable name?

dazlich commented:
Bill Skamarock had it made for me last year.  I will look into it further

Sent from my iPhone

On Sep 8, 2023, at 5:28 PM, gdicker1 ***@***.***> wrote:

﻿

** Caution: EXTERNAL Sender **

@dazlich<https://github.com/dazlich> I've added the ncdata and bnd_topo files you requested but I haven't pushed it yet. I'd like some context on x1.40962.pel.58levels_2000010100_init.nc and to maybe align it better with other ncdata files already in the namelist_defaults.xml.

  *   What's the significance of "pel" in the file name? "PE layout"?
  *   Did you use CFSR data to initialize or some other data source?

Given that, would "cami_01-01-2000_00Z_mpasa120_L58_CFSR_230901.nc" be an acceptable name?

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/CAM/pull/4#issuecomment-1712332504>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWC3WB3NG4NCASIB373XZOSZVANCNFSM6AAAAAA4DTKA4Y>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

dazlich commented:
On Sep 8, 2023, at 5:32 PM, Dazlich,Donald ***@***.***> wrote:

Bill Skamarock had it made for me last year.  I will look into it further

Sent from my iPhone

On Sep 8, 2023, at 5:28 PM, gdicker1 ***@***.***> wrote:

﻿

** Caution: EXTERNAL Sender **


@dazlich<https://github.com/dazlich> I've added the ncdata and bnd_topo files you requested but I haven't pushed it yet. I'd like some context on x1.40962.pel.58levels_2000010100_init.nc and to maybe align it better with other ncdata files already in the namelist_defaults.xml.

  *   What's the significance of "pel" in the file name? "PE layout"?
  *   Did you use CFSR data to initialize or some other data source?

Given that, would "cami_01-01-2000_00Z_mpasa120_L58_CFSR_230901.nc" be an acceptable name?

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/CAM/pull/4#issuecomment-1712332504>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWC3WB3NG4NCASIB373XZOSZVANCNFSM6AAAAAA4DTKA4Y>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


I know there’s some discussion about multiple versions of 58L ic and their provenance. Looking at the global attributes of this file doesn’t give me any clue. If it indeed CFSR I have no problem with renaming it.

// global attributes:
:model_name = "mpas" ;
:core_name = "init_atmosphere" ;
:source = "MPAS" ;
:Conventions = "MPAS" ;
:git_version = "v7.3-370-g25365456-dirty" ;
:on_a_sphere = "YES" ;
:sphere_radius = 6371229. ;
:is_periodic = "NO" ;
:x_period = 0. ;
:y_period = 0. ;
:history = "mpirun -n 36 ./init_atmosphere_model" ;
:parent_id = "dyjep9dspg\n",
"" ;
:mesh_spec = "0.0" ;
:config_init_case = 7 ;
:config_calendar_type = "gregorian" ;
:config_start_time = "2000-01-01_00:00:00" ;
:config_stop_time = "2000-01-01_00:00:00" ;
:config_theta_adv_order = 3 ;
:config_coef_3rd_order = 1. ;
:config_num_halos = 2 ;
:config_nvertlevels = 58 ;
:config_nsoillevels = 4 ;
:config_nfglevels = 38 ;
:config_nfgsoillevels = 4 ;
:config_months = 12 ;
:config_geog_data_path = "/glade/work/wrfhelp/WPS_GEOG/" ;
:config_met_prefix = "ERA5" ;
:config_sfc_prefix = "SST" ;
:config_fg_interval = 86400 ;
:config_landuse_data = "MODIFIED_IGBP_MODIS_NOAH" ;
:config_topo_data = "GMTED2010" ;
:config_vegfrac_data = "MODIS" ;
:config_albedo_data = "MODIS" ;
:config_maxsnowalbedo_data = "MODIS" ;
:config_supersample_factor = 3 ;
:config_use_spechumd = "NO" ;
:config_ztop = 30000. ;
:config_nsmterrain = 0 ;
:config_smooth_surfaces = "YES" ;
:config_dzmin = 0.3 ;
:config_nsm = 30 ;
:config_tc_vertical_grid = "YES" ;
:config_specified_zeta_levels = "levels_L58_i1.txt" ;
:config_blend_bdy_terrain = "NO" ;
:config_extrap_airtemp = "linear" ;
:config_static_interp = "NO" ;
:config_native_gwd_static = "NO" ;
:config_gwd_cell_scaling = 1. ;
:config_vertical_grid = "YES" ;
:config_met_interp = "YES" ;
:config_input_sst = "NO" ;
:config_frac_seaice = "NO" ;
:config_pio_num_iotasks = 0 ;
:config_pio_stride = 1 ;
:config_block_decomp_file_prefix = "x1.40962.graph.info.part." ;
:config_number_of_blocks = 0 ;
:config_explicit_proc_decomp = "NO" ;
:config_proc_decomp_file_prefix = "graph.info.part." ;
:file_id = "0om2o5nbme" ;

gdicker1 commented:
Force push [7fa798e](https://github.com/EarthWorksOrg/CAM/commit/7fa798e00b67f7d97d9166e45115842949f6046f) to [9319a64](https://github.com/EarthWorksOrg/CAM/commit/9319a64dc206bf5490d286d0ba4f1d7392db802a) corrects an `ndata` starting tag to be `ncdata`. Thanks to dazlich for the catch!

dazlich commented:
Shouldn't  we get some 15km defaults defined as well since the new release is supposed to support it?

gdicker1 commented:
That would be good. Do you have a 15km case working that I could grab some filepaths from?

dazlich commented:
These are the files I’m using testing 15km. The atm ic is 58 layers. I don’t have anything 32 layers at 15km. Feel free to rename the ncdata and bnd_topo files. I’m not certain of the provenance of the ncdata file and if there is another it is almost certain be to be preferred, but this gets us something for now.


 bnd_topo = '/glade/work/dazlich/km15/mpas_x1.2621442_cam_topo_file_gmted2010_modis_bedmachine_nc3000_Laplace0015_noleak_20230315.nc'

 ncdata   = '/glade/work/dazlich/km15/x1.2621442_camtopo_full_init.nc'

 drydep_srf_file = '/glade/p/cesmdata/cseg/inputdata/atm/cam/chem/trop_mam/atmsrf_mpasa15_20210804.nc'



 fsurdat = '/glade/p/cesmdata/cseg/inputdata/lnd/clm2/surfdata_map/surfdata_mpasa15_hist_78pfts_CMIP6_simyr2000_c211111.nc'


On Sep 18, 2023, at 12:01 PM, gdicker1 ***@***.***> wrote:


** Caution: EXTERNAL Sender **


That would be good. Do you have a 15km case working that I could grab some filepaths from?

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/CAM/pull/4#issuecomment-1724111772>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWDBUAXPBU2DPH47BFTX3CD6LANCNFSM6AAAAAA4DTKA4Y>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Commit:
Update mpasa120km 58 levels filename and fix typo

Commit:
Add files for mpasa15 hgrid to namelist_defaults_cam.xml

Includes ncdata for 58 levels along with corresponding bnd_topo and
drydep_srf_file

Commit:
Update mpasa15 ncdata filename to match convention

gdicker1 commented:
Alright, I got these files added to namelist_defaults_cam.xml and inputdata2

- bnd_topo: atm/cam/topo/mpas_15_nc3500_c20230315.nc
- ncdata: atm/cam/inic/mpas/cami_01-01-2000_00Z_mpasa15_L58_c230316.nc
- drydep_srf_file: atm/cam/chem/trop_mam/atmsrf_mpasa15_c20210804.nc

The fsurdat file was already added to https://github.com/EarthWorksOrg/CTSM/pull/1

dazlich commented:
Looks like line 343 setting bnd_topo for mpasa15 is incomplete and needs to be terminated with </bnd_topo>



On Sep 19, 2023, at 11:57 AM, gdicker1 ***@***.***> wrote:


** Caution: EXTERNAL Sender **


Alright, I got these files added to namelist_defaults_cam.xml and inputdata2

  *   bnd_topo: atm/cam/topo/mpas_15_nc3500_c20230315.nc
  *   ncdata: atm/cam/inic/mpas/cami_01-01-2000_00Z_mpasa15_L58_c230316.nc
  *   drydep_srf_file: atm/cam/chem/trop_mam/atmsrf_mpasa15_c20210804.nc

The fsurdat file was already added to EarthWorksOrg/CTSM#1<https://github.com/EarthWorksOrg/CTSM/pull/1>

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/CAM/pull/4#issuecomment-1726231565>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWEE6CAH5FEFZWOHAP3X3HMGZANCNFSM6AAAAAA4DTKA4Y>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

dazlich commented:
It would be handy to define mpas_len_disp for 15, 7.5 and 3.75km in the defaults


On Sep 19, 2023, at 11:57 AM, gdicker1 ***@***.***> wrote:


** Caution: EXTERNAL Sender **


Alright, I got these files added to namelist_defaults_cam.xml and inputdata2

  *   bnd_topo: atm/cam/topo/mpas_15_nc3500_c20230315.nc
  *   ncdata: atm/cam/inic/mpas/cami_01-01-2000_00Z_mpasa15_L58_c230316.nc
  *   drydep_srf_file: atm/cam/chem/trop_mam/atmsrf_mpasa15_c20210804.nc

The fsurdat file was already added to EarthWorksOrg/CTSM#1<https://github.com/EarthWorksOrg/CTSM/pull/1>

—
Reply to this email directly, view it on GitHub<https://github.com/EarthWorksOrg/CAM/pull/4#issuecomment-1726231565>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADS3XWEE6CAH5FEFZWOHAP3X3HMGZANCNFSM6AAAAAA4DTKA4Y>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

gdicker1 commented:
> Looks like line 343 setting bnd_topo for mpasa15 is incomplete and needs to be terminated with </bnd_topo>

Sorry looks like I didn't get that into the `cam-ew1.0.004` tag I pushed. Try again now.

> It would be handy to define mpas_len_disp for 15, 7.5 and 3.75km in the defaults

Agreed. I'll add those soon

Commit:
Add mpas_dt and mpas_len_disp for 15, 7.5, and 3.75km grids

Commit:
Add mpasa 15km notopo ncdata

gdicker1 commented:
With the addition of a 32 level 15km notopo file, I think this PR can be wrapped up.
